Modules/Scripted/SampleData/SampleData.py

```diff
@@ -309,7 +316,15 @@ def downloadFromSource(self,source):
     for uri,fileName,nodeName in zip(source.uris,source.fileNames,source.nodeNames):
       filePath = self.downloadFileIntoCache(uri, fileName)
       if nodeName:
-        nodes.append(self.loadNode(filePath, nodeName, source.loadFileType, source.loadFileProperties))
+        loadedNode = self.loadNode(filePath, nodeName, source.loadFileType, source.loadFileProperties)
+        if loadedNode is None:
+          self.logMessage('<b>Load failed! Trying to download again...</b>\n', True)
+          if os.access(filePath, os.W_OK):
+            os.remove(filePath)
+          else:
+            self.logMessage('<b>Load failed! Unable to delete and try again loading %s!</b>\n' % filePath, True)
```

Comment:
The message is reported only if the file doesn't exist not if it couldn't be deleted.
Some exception would have to be caught.  
See https://docs.python.org/2/library/os.html#os.remove

Using the Qt implementation  may have a more robust  behavior across platform.